mllib/src/test/scala/org/apache/spark/ml/feature/StringIndexerSuite.scala

```diff
@@ -47,6 +47,11 @@ class StringIndexerSuite extends SparkFunSuite with MLlibTestSparkContext {
     // a -> 0, b -> 2, c -> 1
     val expected = Set((0, 0.0), (1, 2.0), (2, 1.0), (3, 0.0), (4, 0.0), (5, 1.0))
     assert(output === expected)
+    // convert reverse our transform
+    val reversed = indexer.invert().transform(transformed.select("id", "labelIndex"))
+      .select("id", "label")
+    assert(df.collect().map(r => (r.getInt(0), r.getString(1))).toSet ===
```

Comment:
It'd be nice to check more carefully by zipping the original label column with the newly created one, and checking for equality.